From 63444ae34cb83ef4a51e3a65c9b5d2e58f582b44 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 23 Nov 2023 16:28:48 +0100
Subject: [PATCH 0363/1695] dt-bindings: stm32_tamp: add tamper bindings

Add the internal and external tamper device tree bindings.

While there, add missing st,tamp-id property.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 3fff7fc354e7d5bc2574cbc9c6c3f3bd13638994 dt-bindings: stm32_tamp: add tamper bindings]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I3df4bda65192e83284ee2d55d609a785ff240461
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/343126
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/349410
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 .../dt-bindings/tamper/st,stm32-tamp.h        | 25 +++++++++++++
 .../dt-bindings/tamper/st,stm32mp13-tamp.h    | 34 +++++++++++++++++
 .../dt-bindings/tamper/st,stm32mp25-tamp.h    | 37 +++++++++++++++++++
 3 files changed, 96 insertions(+)
 create mode 100644 core/include/dt-bindings/tamper/st,stm32-tamp.h
 create mode 100644 core/include/dt-bindings/tamper/st,stm32mp13-tamp.h
 create mode 100644 core/include/dt-bindings/tamper/st,stm32mp25-tamp.h

diff --git a/core/include/dt-bindings/tamper/st,stm32-tamp.h b/core/include/dt-bindings/tamper/st,stm32-tamp.h
new file mode 100644
index 000000000..8edfd877b
--- /dev/null
+++ b/core/include/dt-bindings/tamper/st,stm32-tamp.h
@@ -0,0 +1,25 @@
+/* SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-2-Clause) */
+/*
+ * Copyright (C) 2023, STMicroelectronics - All Rights Reserved
+ * Author: Gatien Chevallier <gatien.chevallier@foss.st.com>
+ */
+
+/* Internal Tampers */
+#define INT_TAMPER_RTC_VOLTAGE_MONITORING	1
+#define INT_TAMPER_TEMPERATURE_MONITORING	2
+#define INT_TAMPER_LSE_MONITORING		3
+#define INT_TAMPER_HSE_MONITORING		4
+#define INT_TAMPER_RTC_CALENDAR_OVERFLOW	5
+/* Nothing for tampers 6-7*/
+#define INT_TAMPER_6				6
+#define INT_TAMPER_7				7
+#define INT_TAMPER_MONOTONIC_COUNTER_		8
+
+/* External Tampers */
+#define EXT_TAMPER_1				1
+#define EXT_TAMPER_2				2
+#define EXT_TAMPER_3				3
+
+/* Tamper mode */
+#define TAMPER_CONFIRMED_MODE			1
+#define TAMPER_POTENTIAL_MODE			2
diff --git a/core/include/dt-bindings/tamper/st,stm32mp13-tamp.h b/core/include/dt-bindings/tamper/st,stm32mp13-tamp.h
new file mode 100644
index 000000000..e6709f454
--- /dev/null
+++ b/core/include/dt-bindings/tamper/st,stm32mp13-tamp.h
@@ -0,0 +1,34 @@
+/* SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-3-Clause) */
+/*
+ * Copyright (C) STMicroelectronics 2023 - All Rights Reserved
+ * Author: Gatien Chevallier <gatien.chevallier@foss.st.com>
+ */
+
+/* Internal Tampers */
+#define INT_TAMPER_BKUP_DOMAIN_THRESHOLD	1
+#define INT_TAMPER_TEMPERATURE_MONITORING	2
+#define INT_TAMPER_LSE_MONITORING		3
+#define INT_TAMPER_HSE_MONITORING		4
+#define INT_TAMPER_RTC_CALENDAR_OVERFLOW	5
+#define INT_TAMPER_JTAG_SWD_ACCESS		6
+#define INT_TAMPER_ADC2_AWD_MONITORING_1	7
+#define INT_TAMPER_MONOTONIC_COUNTER_1		8
+#define INT_TAMPER_CRYPTO_PERIPH_FAULT		9
+#define INT_TAMPER_MONOTONIC_COUNTER_2		10
+#define INT_TAMPER_IWDG1_RESET			11
+#define INT_TAMPER_ADC2_AWD_MONITORING_2	12
+#define INT_TAMPER_ADC2_AWD_MONITORING_3	13
+
+/* External Tampers */
+#define EXT_TAMPER_1				1
+#define EXT_TAMPER_2				2
+#define EXT_TAMPER_3				3
+#define EXT_TAMPER_4				4
+#define EXT_TAMPER_5				5
+#define EXT_TAMPER_6				6
+#define EXT_TAMPER_7				7
+#define EXT_TAMPER_8				8
+
+/* Tamper mode */
+#define TAMPER_CONFIRMED_MODE			1
+#define TAMPER_POTENTIAL_MODE			2
diff --git a/core/include/dt-bindings/tamper/st,stm32mp25-tamp.h b/core/include/dt-bindings/tamper/st,stm32mp25-tamp.h
new file mode 100644
index 000000000..c2ffebdbd
--- /dev/null
+++ b/core/include/dt-bindings/tamper/st,stm32mp25-tamp.h
@@ -0,0 +1,37 @@
+/* SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-2-Clause) */
+/*
+ * Copyright (C) 2023, STMicroelectronics - All Rights Reserved
+ * Author: Gatien Chevallier <gatien.chevallier@foss.st.com>
+ */
+
+/* Internal Tampers */
+#define INT_TAMPER_BKUP_DOMAIN_THRESHOLD	1
+#define INT_TAMPER_TEMPERATURE_MONITORING	2
+#define INT_TAMPER_LSE_MONITORING		3
+#define INT_TAMPER_HSE_MONITORING		4
+#define INT_TAMPER_RTC_CALENDAR_OVERFLOW	5
+#define INT_TAMPER_JTAG_SWD_ACCESS		6
+#define INT_TAMPER_VDDCORE_MONITORING		7
+#define INT_TAMPER_MONOTONIC_COUNTER_1		8
+#define INT_TAMPER_CRYPTO_PERIPH_FAULT		9
+#define INT_TAMPER_MONOTONIC_COUNTER_2		10
+#define INT_TAMPER_IWDG3_RESET			11
+#define INT_TAMPER_VDDCPU_MONITORING		12
+/* No event linked to tamper13 */
+#define INT_TAMPER_13				13
+#define INT_TAMPER_IWDG5_RESET			14
+#define INT_TAMPER_IWDG1_RESET			15
+
+/* External Tampers */
+#define EXT_TAMPER_1				1
+#define EXT_TAMPER_2				2
+#define EXT_TAMPER_3				3
+#define EXT_TAMPER_4				4
+#define EXT_TAMPER_5				5
+#define EXT_TAMPER_6				6
+#define EXT_TAMPER_7				7
+#define EXT_TAMPER_8				8
+
+/* Tamper mode */
+#define TAMPER_CONFIRMED_MODE			1
+#define TAMPER_POTENTIAL_MODE			2
-- 
2.47.2

